### PR TITLE
Leany labs/storybook 1 (#1)

### PR DESCRIPTION
**Pull request by LastExileUA**
docs: create new storybooks for Avatar, Chip, Toggle
tests: change style and expend tests for Avatar, Chip, Toggle, Button

### CODE REVIEW

File: src/components/Avatar/__stories__/avatar.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,205 @@
+import { useState, useCallback, useMemo } from "react";
+import Avatar from "../Avatar";
+import { createComponentTemplate } from "../../../storybook/functions/create-component-story";
+import { ArgsTable, Story, Canvas, Meta } from "@storybook/addon-docs";
+import { Calendar, Remove, Add, Check } from "../../Icon/Icons";
+import { TOOLTIP, LABEL, CHIPS } from "../../../storybook/components/related-components/component-description-map";
+import "./avatar.stories.scss";
```
Comment (hadasfa) [inline src/components/Avatar/__stories__/avatar.stories.mdx:7]: please move the scss import to be the last import in the file (that's our convention)

File: src/components/Avatar/__stories__/avatar.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+      size={Avatar.sizes.SMALL}
+      type={Avatar.types.TEXT}
+      text="R"
+      backgroundColor={Avatar.colors.BRIGHT_BLUE}
+      isSquare
+    />
+    <Avatar
+      size={Avatar.sizes.MEDIUM}
+      type={Avatar.types.ICON}
+      text="RM"
```
Comment (hadasfa) [inline src/components/Avatar/__stories__/avatar.stories.mdx:101]: please remove the usage on the text prop since there is no need in it in the Icon avatar

File: src/components/Toggle/__stories__/toggle.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    <Toggle />
+  </Story>
+</Canvas>
+
+### Turn on/ off notifications
+In the automations center, a user can turn the automation on or off.
+<Canvas>
+  <Story name="Turn on/ off an notifications">
+    <b>Do not disturb mode</b>
+    <Toggle ariaControls="asdsad" isDefaultSelected={false} />
```
Comment (hadasfa) [inline src/components/Toggle/__stories__/toggle.stories.mdx:126]: If the toggle does not control anything I think that there is not need to pass ariaControls. What do you think @orrgottlieb ?

File: src/components/Tooltip/__stories__/TooltipContent.jsx [hunk 4]
```diff
@@ -2,7 +2,7 @@ import React from "react";
 import "./TooltipContent.scss";
 
 const TooltipContent = () => {
-  return <div>Tooltip Content</div>;
+  return <div>Text</div>;
```
Comment (hadasfa) [inline src/components/Tooltip/__stories__/TooltipContent.jsx:5]: Why we do this change?

File: src/storybook/components/related-components/descriptions/checkbox-description.jsx [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import { useMemo } from "react";
+import { RelatedComponent } from "../../related-component/related-component";
+import { Checkbox } from "../../../../components/Checkbox/Checkbox";
+
+export const CheckboxDescription = () => {
+  const component = useMemo(() => <Checkbox label="Selected" checked={true} />, []);
```
Comment (hadasfa) [inline src/storybook/components/related-components/descriptions/checkbox-description.jsx:6]: please use shorthand syntax for set a true (boolean) value to a prop, such in the following example: `<Checkbox label="Selected" checked />,`

File: src/storybook/components/related-components/descriptions/chips-description.jsx [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import { useMemo } from "react";
+import { RelatedComponent } from "../../related-component/related-component";
+import Chips from "../../../../components/Chips/Chips";
+
+export const ChipsDescription = () => {
+  const component = useMemo(() => <Chips label="This is a chip" readOnly={true} />, []);
```
Comment (hadasfa) [inline src/storybook/components/related-components/descriptions/chips-description.jsx:6]: please use shorthand syntax for set a true (boolean) value to a prop, such in the following example: `<Checkbox label="Selected" checked />,`

File: src/storybook/components/related-components/descriptions/counter-description.jsx [hunk 7]
```diff
@@ -0,0 +1,10 @@
+import { useMemo } from "react";
+import { RelatedComponent } from "../../related-component/related-component";
+import Counter from "../../../../components/Counter/Counter";
+
+export const CounterDescription = () => {
+  const component = useMemo(() => <Counter count="5" />, []);
```
Comment (hadasfa) [inline src/storybook/components/related-components/descriptions/counter-description.jsx:6]: please pass a numeric value to the "count" prop of the counter (since the "count" prop type is numeric)

File: src/storybook/components/related-components/descriptions/radio-button-description.jsx [hunk 8]
```diff
@@ -0,0 +1,14 @@
+import { useMemo } from "react";
+import { RelatedComponent } from "../../related-component/related-component";
+import RadioButton from "../../../../components/RadioButton/RadioButton";
+
+export const RadioButtonDescription = () => {
+  const component = useMemo(() => <RadioButton text="Selection" checked={true} />, []);
```
Comment (hadasfa) [inline src/storybook/components/related-components/descriptions/radio-button-description.jsx:6]: please use shorthand syntax for setting true boolean value for a prop `<RadioButton text="Selection" checked />`.

File: src/components/Button/__stories__/button.stories.mdx [hunk 9]
```diff
@@ -228,4 +228,4 @@ When one action is selected, the other is disabled
 </Canvas>
 
 ## Related components
-<RelatedComponents componentsNames={[SPLIT_BUTTON, BUTTON_GROUP, BUTTON_GROUP]}/>
\ No newline at end of file
+<RelatedComponents componentsNames={[SPLIT_BUTTON, BUTTON_GROUP, CHIPS]}/>
```
Comment (hadasfa) [inline src/components/Button/__stories__/button.stories.mdx:231]: let's change the name of the "chips" constant to CHIP: "chip" because in the near future we will change the component name (please do not change the component name for now, only the constant and the name of the component description component)

File: src/components/Button/__tests__/button-tests.jest.js [hunk 10]
```diff
@@ -1,14 +1,15 @@
 import React from "react";
-import { render, fireEvent, cleanup, act } from "@testing-library/react";
-import { sinon, expect } from "../../test/test-helpers";
-import Button from "./Button";
+import { render, fireEvent, cleanup } from "@testing-library/react";
+import { sinon, expect } from "../../../test/test-helpers";
```
Comment (hadasfa) [inline src/components/Button/__tests__/button-tests.jest.js:3]: please revert the changes for the button component tests (we do not want to do any changes in this component tests right now)

File: src/components/Chips/__tests__/chips-snapshot-tests.jest.js [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+    expect(tree).toMatchSnapshot();
+  });
+
+  it("renders correctly with color", () => {
+    const tree = renderer.create(<Chips color={Chips.colors.NEGATIVE} />).toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+  
+  it("renders correctly disabled chip", () => {
+    const tree = renderer.create(<Chips disabled={true} />).toJSON();
```
Comment (hadasfa) [inline src/components/Chips/__tests__/chips-snapshot-tests.jest.js:18]: please change the syntax when setting a true boolean value to a prop to short hand syntax like this: `<Chips disabled} />`

File: src/components/Chips/__tests__/chips-snapshot-tests.jest.js [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+    const tree = renderer.create(<Chips readOnly={true} />).toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+
+  it("renders correctly with text", () => {
+    const tree = renderer.create(<Chips label="text" />).toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+
+  it.skip("renders correctly with right icon ", () => {
```
Comment (hadasfa) [inline src/components/Chips/__tests__/chips-snapshot-tests.jest.js:32]: why we skip this test? and let's add a test for left icon too.

File: src/components/Chips/__tests__/chips-tests.jest.js [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+    );
+  });
+
+  it("should call the click callback when clicked", () => {
+    fireEvent.click(screen.queryByRole('button'));
+    sinonExpect(onDeletedStub).to.be.calledOnce;
+  })
+
+  it("should not have close icon when read only", () => {
+    const { container, rerender } = chipComponent;
```
Comment (hadasfa) [inline src/components/Chips/__tests__/chips-tests.jest.js:25]: I think we can remove this check (it's enough to check it in the snapshot only). @orrgottlieb  what do you think?

File: src/components/Chips/__tests__/chips-tests.jest.js [hunk 14]
```diff
@@ -0,0 +1,35 @@
...
+    rerender (<Chips readOnly />);
+    expect(container.firstChild.classList.contains('with-close')).toBeFalsy();
+  })
+
+  it("should not have close icon when disabled", () => {
```
Comment (hadasfa) [inline src/components/Chips/__tests__/chips-tests.jest.js:30]: I think we can remove this check (it's enough to check it in the snapshot only). @orrgottlieb what do you think?

File: src/components/Toggle/__tests__/toggle-snapshot-tests.jest.js [hunk 15]
```diff
@@ -0,0 +1,36 @@
+import React from "react";
+import renderer from "react-test-renderer";
+import Toggle from "../Toggle";
+
```
Comment (hadasfa) [inline src/components/Toggle/__tests__/toggle-snapshot-tests.jest.js:4]: let's add here also checks for   ariaLabel,
  ariaControls,
  areLabelsHidden,
  onOverrideText,
  offOverrideText

File: src/components/Toggle/__tests__/toggle-tests.jest.js [hunk 16]
```diff
@@ -0,0 +1,95 @@
...
+  describe("Default selected mode", () => {
+
+    afterEach(() => {
+      cleanup();
+    });
+
+    it("should change state to off when is default selected and clicked", () => {
+      const { getByRole } = render(
+        <form name={formName}>
+          <Toggle componentClassName="dummy-class-name" isDefaultSelected />
```
Comment (hadasfa) [inline src/components/Toggle/__tests__/toggle-tests.jest.js:21]: no need here for setting componentClassName (not relevant to the test)

File: src/components/Toggle/__tests__/toggle-tests.jest.js [hunk 17]
```diff
@@ -0,0 +1,95 @@
...
+      );
+
+      const toggle = getByRole(toggleRole);
+      userEvent.click(toggle);
+      expect(toggle.checked).toBeTruthy();
+    });
+  });
+
+  describe("a11y", () => {
+    it("should add the aria label", () => {
```
Comment (hadasfa) [inline src/components/Toggle/__tests__/toggle-tests.jest.js:55]: I think we should check the existence of a11y props by snapshot tests. what do you think? @orrgottlieb

File: src/components/Chips/__tests__/chips-tests.jest.js (hunk 13 above)
Comment (orrgottlieb) [reply src/components/Chips/__tests__/chips-tests.jest.js:25]: i agree

File: src/components/Chips/__tests__/chips-tests.jest.js [hunk 18]
```diff
@@ -0,0 +1,35 @@
...
+import { sinon, expect as sinonExpect } from "../../../test/test-helpers";
+
+describe("Chips tests", () => {
+  const lable = "Chip";
+  const className = "test-class";
+  let onDeletedStub;
+  let chipComponent;
+
+  beforeEach(() => {
+    onDeletedStub = sinon.stub();
```
Comment (orrgottlieb) [inline src/components/Chips/__tests__/chips-tests.jest.js:13]: please use `jest.fn()`

File: src/components/Toggle/__stories__/toggle.stories.mdx [hunk 19]
```diff
@@ -0,0 +1,131 @@
...
+  ]}
+/>
+
+## Use cases and examples
+
+### Turn on/ off an automation
+In the automations center, a user can turn the automation on or off.
+<Canvas>
+  <Story name="Turn on/ off an automation">
+    <b>Board automations</b>
```
[the remark this answers is not in the document]
Comment (orrgottlieb) [reply src/components/Toggle/__stories__/toggle.stories.mdx:116]: we have Paragraph bold class

File: src/components/Toggle/__stories__/toggle.stories.mdx (hunk 3 above)
Comment (orrgottlieb) [reply src/components/Toggle/__stories__/toggle.stories.mdx:126]: i agree

File: src/components/Toggle/__tests__/toggle-tests.jest.js (hunk 17 above)
Comment (orrgottlieb) [reply src/components/Toggle/__tests__/toggle-tests.jest.js:55]: I disagree, it depends. when you can simulate interactivity via a11y props so it will simulated screen reader interactibilty

**Review by orrgottlieb — CHANGES_REQUESTED** [review]

File: src/components/Toggle/__tests__/toggle-tests.jest.js (hunk 17 above)
Comment (hadasfa) [reply src/components/Toggle/__tests__/toggle-tests.jest.js:55]: So let's keep the aria props tests like this

File: src/components/Tooltip/__stories__/TooltipContent.jsx (hunk 4 above)
Comment (LastExileUA) [reply src/components/Tooltip/__stories__/TooltipContent.jsx:5]: This component is only used inside Storybook, and in the new design, it contains "Text".

**Review by hadasfa — APPROVED** [review]

**Comment by hadasfa** [comment]
Amazing job! Thank you!
